packages/metadata-common/src/FilterTools.tsx

```diff
@@ -134,16 +136,17 @@ export class FilterTools extends React.Component<
         className={`${FILTER_TAG} tag unapplied-tag`}
         id={'filter' + '-' + tag + '-' + index}
         key={'filter' + '-' + tag + '-' + index}
+        title={tag}
         onClick={this.handleClick}
       >
-        {tag}
+        <span className={FILTER_TAG_LABEL}>{tag}</span>
       </button>
     );
   }
 
   handleClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void {
     const target = event.target as HTMLElement;
-    const clickedTag = target.innerText.trim();
+    const clickedTag = target.textContent ?? target.innerHTML;
```

Comment:
Maybe using just `target.textContent` would solve the issue here, I don't see a case it would fall into `target.innerHTML`

Now that you pointed out to the whitespace bug I think we should clean whitespaces from both ends of a text when user registers a new tag. This can be a separate issue.

Reply:
For textContent it can be null so I had to return a non null value in the case it is null. I think I could also do "" instead of target.innerHTML.

Reply:
Ok, yeah maybe empty string as a fallback might be better then, otherwise `innerHTML` will always resolve to the new span element
`<span class=\"elyra-filter-tag-label\"> whitespace</span>`

Reply:
okay I changed it to empty string